src/Microsoft.Health.Dicom.Api/Logging/DicomTelemetryInitializer.cs

```diff
@@ -0,0 +1,54 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using EnsureThat;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Health.Dicom.Api.Features.Telemetry;
+
+namespace Microsoft.Health.Dicom.Api.Logging;
+
+public class DicomTelemetryInitializer : ITelemetryInitializer
+{
+    private readonly IHttpContextAccessor _httpContextAccessor;
+
+    public DicomTelemetryInitializer(IHttpContextAccessor httpContextAccessor)
+    {
+        _httpContextAccessor = EnsureArg.IsNotNull(httpContextAccessor, nameof(httpContextAccessor));
+    }
+
+    public void Initialize(ITelemetry telemetry)
+    {
+        if (telemetry is RequestTelemetry requestTelemetry)
+            AddPropertiesFromHttpContextItems(requestTelemetry);
```

Comment:
Will this step gets executed for all the health check requests too?

Reply:
That is just some extra unwanted processing, since health check happens every 15-30 seconds. You can keep it if you need these columns to be populated in RequestTelemetry for health checks.